src/lib/sed.c

```diff
@@ -298,46 +290,42 @@ int sed_eraselr(struct sed_device *dev, const char *password,
 	return curr_if->eraselr_fn(dev, password, key_len, user, lr, sum);
 }
 
-int sed_ds_admin_write(struct sed_device *dev, const char *key, uint8_t key_len,
-		const void *from, uint32_t size, uint32_t offset)
+int sed_ds_read(struct sed_device *dev, enum SED_AUTHORITY auth,
+		const struct sed_key *key, uint8_t *to, uint32_t size,
+		uint32_t offset)
 {
-	if (curr_if->ds_admin_write_fn == NULL)
+	if (curr_if->ds_read_fn == NULL)
 		return -EOPNOTSUPP;
 
-	return curr_if->ds_admin_write_fn(dev, key, key_len, from, size, offset);
-}
-
-int sed_ds_admin_read(struct sed_device *dev, const char *key, uint8_t key_len,
-		uint8_t *to, uint32_t size, uint32_t offset)
-{
-	if (curr_if->ds_admin_read_fn == NULL)
-		return -EOPNOTSUPP;
+	if (auth != SED_ANYBODY && key == NULL) {
+		SEDCLI_DEBUG_MSG("Key can't be null\n");
+		return -EINVAL;
+	}
 
-	return curr_if->ds_admin_read_fn(dev, key, key_len, to, size, offset);
+	return curr_if->ds_read_fn(dev, auth, key, to, size, offset);
 }
 
-int sed_ds_anybody_read(struct sed_device *dev, uint8_t *to, uint32_t size, uint32_t offset)
+int sed_ds_write(struct sed_device *dev, enum SED_AUTHORITY auth,
+		const struct sed_key *key, const void *from, uint32_t size,
+		uint32_t offset)
 {
-	if (curr_if->ds_anybody_read_fn == NULL)
+	if (curr_if->ds_write_fn == NULL)
 		return -EOPNOTSUPP;
 
-	return curr_if->ds_anybody_read_fn(dev, to, size, offset);
-}
-
-int sed_ds_anybody_write(struct sed_device *dev, uint8_t *from, uint32_t size, uint32_t offset)
-{
-	if (curr_if->ds_anybody_write_fn == NULL)
-		return -EOPNOTSUPP;
+	if (auth != SED_ANYBODY && key == NULL) {
+		SEDCLI_DEBUG_MSG("Key can't be null\n");
```

Comment:
How about, "Key can't be NULL for this authority"..? Just to be more clear.
But this is fine as well